### PR TITLE
`isPvEncryptionInTransitEnabled` not being set properly in `OCIMachine`

### LINKED ISSUE #245: `isPvEncryptionInTransitEnabled` not being set properly in `OCIMachine`
**What happened**:
When using the `cluster-template.yaml` and setting the ENV `OCI_NODE_PV_TRANSIT_ENCRYPTION=false` I noticed the `isPvEncryptionInTransitEnabled` is still set to true when trying to launch an instance. The `OCIMachineTemplate` shows `isPvEncryptionInTransitEnabled` as `false` but the `OCIMachine` is `true`.

**What you expected to happen**:
I would expect the `OCIMachine` to show the value of `isPvEncryptionInTransitEnabled` as `false` and use that same value when trying to launch an instance.

**How to reproduce it (as minimally and precisely as possible)**:
Use a build with commit f620d39e8abc67c531e8bff468ef5989432143c4 or after of CAPOCI. Set the ENV `OCI_NODE_PV_TRANSIT_ENCRYPTION=false` then generate and apply the template using `cluster-template.yaml`.

**Anything else we need to know?**:
I've tested this same flow using commit ee5de6b2045eb564a096a946d8e961b443bd7dab and the value of `isPvEncryptionInTransitEnabled` is `false` as expected for both `OCIMachineTemplate` and `OCIMachine`.

**Environment:**
OSX using tilt in development environment

- CAPOCI version: commit f620d39e8abc67c531e8bff468ef5989432143c4 or after
- Cluster-API version (use `clusterctl version`): 1.4.1
- Kubernetes version (use `kubectl version`): v1.24.4
- Docker version (use `docker info`): 20.10.18 rancher desktop
- OS (e.g. from `/etc/os-release`): OSX 13.3

### PR DESCRIPTION
**Pull request by shyamradhakrishnan**
**What this PR does / why we need it**:
`isPvEncryptionInTransitEnabled` not being set properly in `OCIMachine`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #245

### CODE REVIEW

**Comment by shyamradhakrishnan** [comment]
baremetal e2e test

```
Workload cluster creation when Bare Metal workload cluster creation Bare Metal - With 1 control-plane nodes and 1 worker nodes
/home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci/test/e2e/cluster_test.go:400

[SynchronizedAfterSuite] PASSED [0.912 seconds]
[SynchronizedAfterSuite]
/home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci/test/e2e/e2e_suite_test.go:252

  Timeline >>
  STEP: Tearing down the management cluster @ 04/13/23 10:38:23.264
  << Timeline
------------------------------

Ran 1 of 25 Specs in 1664.330 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 24 Skipped
```

**Comment by shyamradhakrishnan** [comment]
e2e test, the CCM test will run only if the instance boot volume encryption is turned on as we mount para virtualized volumes
```

Ran 6 of 25 Specs in 2919.601 seconds
SUCCESS! -- 6 Passed | 0 Failed | 0 Pending | 19 Skipped


Ginkgo ran 1 suite in 48m46.237469558s
Test Suite Passed
make[1]: Leaving directory '/home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci'
```

File: config/crd/bases/infrastructure.cluster.x-k8s.io_ocimachinepools.yaml [hunk 1]
```diff
@@ -936,7 +936,6 @@ spec:
                         type: string
                     type: object
                   isPvEncryptionInTransitEnabled:
-                    default: true
```
Comment (joekr) [inline config/crd/bases/infrastructure.cluster.x-k8s.io_ocimachinepools.yaml:939]: Just thinking out loud: What happens if kubebuilder sets this to false as default so the CRD still defines a default? Not saying it has to be done, but having a default in the CRD is nicer for the user. 🤷‍♂️

**Review by joekr — APPROVED** [review]

File: config/crd/bases/infrastructure.cluster.x-k8s.io_ocimachinepools.yaml (hunk 1 above)
Comment (shyamradhakrishnan) [reply config/crd/bases/infrastructure.cluster.x-k8s.io_ocimachinepools.yaml:939]: I think or bool, it is generally understood that false is default, our provider and other providers also uses this, so I think this is ok.

**Pull request merged**